addons/l10n_cl_website_sale/models/sale.py

```diff
@@ -0,0 +1,20 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def _get_latam_document_type_by_partner(self, partner_id):
+        latam_document = ''
+        if partner_id.l10n_cl_sii_taxpayer_type == '1':
+            latam_document = self.env.ref('l10n_cl.dc_a_f_dte').id
+        elif partner_id.l10n_cl_sii_taxpayer_type == '3':
+            latam_document = self.env.ref('l10n_cl.dc_b_f_dte').id
+        return latam_document
+
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        res = super()._create_invoices(grouped, final, date)
+        if self.website_id and self.website_id.company_id.country_id.code == 'CL':
+            for invoice in res:
+                invoice.l10n_latam_document_type_id = self._get_latam_document_type_by_partner(invoice.partner_id)
```

Comment:
Does not that happen automatically with the compute?  We do not have it for AR, so why is it needed here? Do not dc_a_f_dte and dc_b_f_dte already correspond to the document type by default?

Reply:
@jco-odoo in CL we are giving the customer the option to select the document type. Besides, we need the document ref id there, because the taxpayer type, does not have a default document. Instead, it defines a domain.